src/auth.ts

```diff
@@ -1,6 +1,33 @@
 import { User } from './config_types';
 
-export interface Authenticator {
-    isAuthProvider(user: User): boolean;
-    getToken(user: User): string | null;
+interface CredentialsBase {
+    expirationTimestamp?: string;
+}
+
+export interface TokenCredentials extends CredentialsBase {
```

Comment:
This should be singular. It's only a single token. `TokenCredential`

Reply:
Sure. Colloquially the plural is pretty widely used, so I just went with that, but I see now that the singular is used elsewhere in the API. Will change.